src/sentry/interfaces/stacktrace.py

```diff
@@ -691,9 +693,15 @@ def to_python(cls, data, slim_frames=True, raw=False):
         if not is_valid:
             raise InterfaceValidationError("Invalid stack frame data.")
 
+        # trim down the frame list for excessive things
+        frameiter = data['frames']
+        if len(data['frames']) > settings.SENTRY_STACKTRACE_FRAMES_HARD_LIMIT:
+            frameiter = chain(
+                islice(data['frames'], settings.SENTRY_STACKTRACE_FRAMES_HARD_LIMIT - 1), (data['frames'][-1],))
```

Comment:
Why does this need the `chain`? Can't this just be

```py
if len(data['frames']) > settings.SENTRY_STACKTRACE_FRAMES_HARD_LIMIT:
  frameiter = islice(data['frames'], settings.SENTRY_STACKTRACE_FRAMES_HARD_LIMIT)
```

maybe an off by one somewhere in there though